nextcord/application_command.py

```diff
@@ -1659,7 +1664,25 @@ async def handle_value(
         elif self.type is ApplicationCommandOptionType.user:
             user_id = int(value)
             user_dict = {user.id: user for user in get_users_from_interaction(state, interaction)}
-            value = user_dict[user_id]
+            try:
+                value = user_dict[user_id]
+            except KeyError:
+                # By here the interaction data doesn't contain
+                # a full member/user object yet so fall back to bot cache
+                value = None
+                data = cast(ApplicationCommandInteractionData, interaction.data)
+                if guild_id := data.get("guild_id"):
+                    if guild := state._guilds.get(int(guild_id)):
+                        value = guild.get_member(user_id)
+
+                if value is None:
+                    # Either we aren't in a guild or
+                    # the member object is not cached
+                    value = state._users.get(user_id)
+
+                if value is None:
+                    # Fall back to a Object at-least
+                    value = Object(id=user_id)
```

Comment:
```suggestion
                    value = state._users.get(user_id)

                    if value is None:
                        # Fall back to a Object at-least
                        value = Object(id=user_id)
```

Check unnecessary if first if was False